ReachabilityUI/ReachabilityUI/Reachability/ReachabilityViewController.swift

```diff
@@ -82,34 +99,48 @@ class ReachabilityViewController: UIViewController {
     // MARK: - Helpers
     
     private func adjustLabelBasedOnConfiguration(_ state: State) {
-        titleLabel.font = configuration.font
-        titleLabel.textAlignment = configuration.textAlignment
+        titleLabel.font = configuration.options[.font] as? UIFont
+        titleLabel.textAlignment = configuration.options[.textAlignment] as! NSTextAlignment
         switch state {
         case .show:
             titleLabel.text = configuration.noConnectionTitle
-            titleLabel.textColor = configuration.noConnectionTitleColor
-            view.backgroundColor = configuration.noConnectionBackgroundColor
+            titleLabel.textColor = configuration.options[.noConnectionTitleColor] as? UIColor
+            view.backgroundColor = configuration.options[.noConnectionBackgroundColor] as? UIColor
         case .hide:
             titleLabel.text = configuration.title
-            titleLabel.textColor = configuration.titleColor
-            view.backgroundColor = configuration.backgroundColor
+            titleLabel.textColor = configuration.options[.titleColor] as? UIColor
+            view.backgroundColor = configuration.options[.backgroundColor] as? UIColor
         }
     }
     
     private func animatePositionChange(_ state: State, animation: ReachabilityConfiguration.Animation) {
+        let height = self.configuration.options[.height] as! CGFloat
```

Comment:
this func clocks in at around 70 lines! That is a lot for a function 😄 

Could parts of it be extracted to private helper methods?